Added rating to performers and studios

julien0221 opened the pull request:
I added rating on performers and studios page. And I added some tests to it.

Resolves #1283 
Resolves #1289

WithoutPants commented on ui/v2.5/src/components/Performers/PerformerDetails/PerformerEditPanel.tsx:
```diff
@@ -62,7 +63,7 @@ export const PerformerEditPanel: React.FC<IPerformerDetails> = ({
   const [scraper, setScraper] = useState<GQL.Scraper | undefined>();
   const [newTags, setNewTags] = useState<GQL.ScrapedSceneTag[]>();
-
+  const [rating, setRating] = useState<number>(performer?.rating ?? NaN);
```
Should be using formik values instead.

This use of this means that if you navigate away from the edit page after changing the rating but not saving, then no prompt is displayed about unsaved data.

WithoutPants reviewed (changes requested):
* setting a rating on a Performer that already is marked as a favourite results in overlapping ribbons in the Performers page:
![image](https://user-images.githubusercontent.com/53250216/115651999-b1865400-a36f-11eb-9637-17557f264806.png)

My suggestion would be to change the favourite ribbon to be a heart icon on the upper right of the performer image.

* rating is not set when creating a performer

julien0221 commented:
I removed the banner favourite and I added a heart icon on upper right as you suggested.
I changed the setRating to use formik to show the unsaved popup.

I created some performers and I am not sure what you mean with "rating is not set when creating a performer". Because it works all good when I was testing.
Can you give me more details please?

WithoutPants reviewed (approved):
The create bug was fixed with the change to use formik values.

WithoutPants commented:
@julien0221 please contact me on Discord to arrange bounty payment.

GsMumbo commented:
@julien0221 @WithoutPants It looks like Star Ratings were added to both, but only Performers can be sorted by rating. We’re Studios also supposed to have this sorting too?

(Also thank you! This is amazing!)
